### ISSUE TITLE
confusion about the training process

### ISSUE THREAD

**Issue by nieve01** [RicardooYoung/VersatileTeacher#7]
Hello, thank you for your work！ I am confused about the training process, I wonder if you can answer it.
1) First, use the source-domain data to train a standard yolov5 model, and then use this model parameters to initialize the teacher model and student model.
2) Then start the second stage of training process, in which both the teacher model and the student model exist. The student model updates the parameters by gradient descent, and the teacher model updates the parameters using the EMA of the student model. So in the end, do you keep the teacher model? In other words, do the test on the teacher model?

**Comment by nieve01** [comment]
python train.py \
    --cfg models/yolov5l_da.yaml \
    --cfg models/yolov5l.yaml \
    --weights weights/yolov5l_cityscapes_joint.pt \
    --weights weights/yolov5l_cityscapes_joint.pt \
    --data data/cityscapes.yaml \
    --data data/cityscapes_foggy.yaml \
    --device 0 \
    --epochs 1 \
    --batch-size 16 \
    --name test

And the first cfg parameter means the student model and the second cfg parameter means the teacher model?

**Comment by RicardooYoung** [comment]
Hello,

Step 1 uses only one model that trained on both domains to obtain a pretrained weight. Step 2 uses the weight to initialize the teacher model and the student model in VT framework and continues training. Actually, some other works (such as AT) combine these two steps, the teacher model is created and copies the weight of the student model after a certain epoch. I decoupled them for the convenience of comparative experiments.

At the end of training, the student model is saved, and testing is conducted on the student model.

As for the cfg parameter, your interpretation is correct. The first is parsed as the student model, and the second as the teacher model.

Hope this helps.

**Comment by nieve01** [comment]
Thanks for your answers！ But I still have some doubts about step 1. Step 1 uses only one model that trained on both domains to obtain a pretrained weight. But aren’t the images of the target domain unlabeled? How does the target domain data participate in the training in step 1?

**Comment by RicardooYoung** [comment]
In Step 1, when processing data from the target domain, only the adversarial loss is enabled. This aims to initialize the domain discriminators.

**Comment by nieve01** [comment]
> In Step 1, when processing data from the target domain, only the adversarial loss is enabled. This aims to initialize the domain discriminators.

thank you !
